tests/unit/externalRequests.js

```diff
@@ -17,8 +17,249 @@ const handleRequestError = externalRequests.__get__('handleRequestError')
+      const status = 202
+
+      const valid = validateRequestStatusCode(allowedStatuses)(status)
+      t.equals(valid, true)
+      t.end()
+    })
+  })
+
+  t.test('performRequests', {autoend: true}, t => {
+    t.test('should create orchestration object', t => {
+      const ctx = {}
+      const requests = []
+
+      performRequests(requests, ctx)
+      t.equals(!ctx.orchestrations, false)
+      t.end()
+    })
+
+    t.test('should create orchestrations array', t => {
+      const ctx = {}
+      const requests = []
+
+      performRequests(requests, ctx)
+      t.equals(!ctx.orchestrations, false)
+      t.end()
+    })
+
+    t.test(
+      'should throw error and create orchestration when one lookup fails( request to server that is down)',
+      async t => {
+        const url = 'http://localhost:8000/'
+        const url2 = 'http://localhost:9000/'
+        const method = 'PUT'
+        const id = '1233243'
+
+        const requests = [
+          {
+            config: {
+              url: url,
+              method: method
+            },
+            id: id
+          },
+          {
+            config: {
+              url: url2,
+              method: method
+            },
+            id: id
+          }
+        ]
+
+        const ctx = {
+          state: {
+            metaData: {
+              requests: {}
+            }
+          },
+          request: {
+            header: {
+              'x-openhim-transactionid': '1232244'
+            }
+          }
+        }
+
+        try {
+          await Promise.all(performRequests(requests, ctx))
+        } catch (error) {
+          t.equals(ctx.orchestrations.length, 1)
+          t.match(error.message, /ECONNREFUSED/)
+          t.end()
+        }
+      }
+    )
+
+    t.test('should do lookups and create orchestrations', async t => {
+      const url = 'http://localhost:4000/'
+
+      nock(url)
+        .get('/patient')
+        .reply(200, 'Body')
+
+      const method = 'GET'
+      const id = '123'
+
+      const requests = [
+        {
+          config: {
+            url: `${url}patient`,
+            method: method
+          },
+          id: id
+        }
+      ]
+
+      const ctx = {
+        state: {
+          metaData: {
+            requests: {}
+          }
+        },
+        request: {
+          header: {
+            'x-openhim-transactionid': '1232244'
+          }
+        }
+      }
+
+      await Promise.all(performRequests(requests, ctx)).then(res => {
+        t.deepEqual(res[0], {'123': 'Body'})
+        t.equals(ctx.orchestrations.length, 1)
+        t.end()
+      })
+    })
+  })
+
+  t.test('prepareLookupRequests', {autoend: true}, t => {
+    t.test(
+      'should not do any lookups when requests metadata object is empty',
+      t => {
+        const ctx = {
+          state: {
+            metaData: {}
+          }
+        }
+        prepareLookupRequests(ctx)
+
+        t.equals(!ctx.orchestrations, true)
```

Comment:
This tap assertion is better suited to what you're trying to test.
Anyway, adding the `not` version of an assertion is easier to understand than adding `!` to the logic

```suggestion
        t.notOk(ctx.orchestrations)
```